x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java

```diff
@@ -275,8 +275,8 @@ public void onResponse(Boolean mappingsUpdated) {
             public void onFailure(Exception e) {
                 if (ExceptionsHelper.unwrapCause(e) instanceof IllegalArgumentException) {
                     // the underlying error differs depending on which way around the clashing fields are seen
-                    Matcher matcher = Pattern.compile("(?:mapper|Can't merge a non object mapping) \\[(.*)\\] (?:cannot be changed " +
-                            "from type \\[.*\\] to|with an object mapping) \\[.*\\]").matcher(e.getMessage());
+                    Matcher matcher = Pattern.compile("Can't merge a non object mapping \\[(.*)\\] with an object mapping")
+                        .matcher(e.getMessage());
```

Comment:
Yikes that we are doing regex matches on exception messages here :( it feels like there should be a better way, but totally out of scope for this PR

Reply:
+1, this was painful to track down and fix.